src/pubsub/ua_pubsub_ns0.c

```diff
@@ -595,6 +648,108 @@ removeDataSetFolderAction(UA_Server *server,
+
+    UA_NodeId selectedFieldsNode =
+        findSingleChildNode(server, UA_QUALIFIEDNAME(0, "SelectedFields"),
+                            UA_NODEID_NUMERIC(0, UA_NS0ID_HASPROPERTY),
+                            UA_NODEID_NUMERIC(0, publishedDataSet->identifier.identifier.numeric));
+    if(UA_NodeId_equal(&selectedFieldsNode, &UA_NODEID_NULL))
+        return UA_STATUSCODE_BADNOTFOUND;
+
+    UA_NodePropertyContext * selectedFieldsContext = (UA_NodePropertyContext *)
+        UA_malloc(sizeof(UA_NodePropertyContext));
+    selectedFieldsContext->parentNodeId = publishedDataSet->identifier;
+    selectedFieldsContext->parentClassifier = UA_NS0ID_PUBLISHEDEVENTSTYPE;
+    selectedFieldsContext->elementClassiefier = UA_NS0ID_PUBLISHEDEVENTSTYPE_SELECTEDFIELDS;
+    retVal |= addVariableValueSource(server, valueCallback, selectedFieldsNode,
+                                     selectedFieldsContext);
+
+    UA_NodeId eventNotifierNode =
+        findSingleChildNode(server, UA_QUALIFIEDNAME(0, "EventNotifier"),
+                            UA_NODEID_NUMERIC(0, UA_NS0ID_HASPROPERTY),
+                            UA_NODEID_NUMERIC(0, publishedDataSet->identifier.identifier.numeric));
+    if(UA_NodeId_equal(&eventNotifierNode, &UA_NODEID_NULL))
+        return UA_STATUSCODE_BADNOTFOUND;
+
+    UA_NodePropertyContext * eventNotifierContext = (UA_NodePropertyContext *)
+        UA_malloc(sizeof(UA_NodePropertyContext));
+    eventNotifierContext->parentNodeId = publishedDataSet->identifier;
+    eventNotifierContext->parentClassifier = UA_NS0ID_PUBLISHEDEVENTSTYPE;
+    eventNotifierContext->elementClassiefier = UA_NS0ID_PUBLISHEDEVENTSTYPE_PUBSUBEVENTNOTIFIER;
+    retVal |= addVariableValueSource(server, valueCallback, eventNotifierNode,
+                                     eventNotifierContext);
+
+    UA_NodeId filterNode =
+        findSingleChildNode(server, UA_QUALIFIEDNAME(0, "Filter"),
+                            UA_NODEID_NUMERIC(0, UA_NS0ID_HASPROPERTY),
+                            UA_NODEID_NUMERIC(0, publishedDataSet->identifier.identifier.numeric));
+    if(UA_NodeId_equal(&filterNode, &UA_NODEID_NULL))
+        return UA_STATUSCODE_BADNOTFOUND;
+
+    UA_NodePropertyContext * filterContext = (UA_NodePropertyContext *)
+        UA_malloc(sizeof(UA_NodePropertyContext));
+    filterContext->parentNodeId = publishedDataSet->identifier;
+    filterContext->parentClassifier = UA_NS0ID_PUBLISHEDEVENTSTYPE;
+    filterContext->elementClassiefier = UA_NS0ID_PUBLISHEDEVENTSTYPE_FILTER;
+    retVal |= addVariableValueSource(server, valueCallback, filterNode,
+                                     filterContext);
+
+#ifdef UA_ENABLE_PUBSUB_INFORMATIONMODEL_METHODS
+    retVal |= UA_Server_addReference(server, publishedDataSet->identifier,
+                                     UA_NODEID_NUMERIC(0, UA_NS0ID_HASCOMPONENT),
+                                     UA_EXPANDEDNODEID_NUMERIC(0, UA_NS0ID_PUBLISHEDEVENTSTYPE_MODIFYFIELDSELECTION), true);
```

Comment:
I think you currently did not add the implementation of the 'MODIFYFIELDSELECTION' function, right?

Reply:
Right, it's still on the todo list. I commented it out with a todo notice. Probably makes sense to change the event config from DSF to PDS before implementing this method.